python/ray/train/session.py

```diff
@@ -289,6 +284,11 @@ def train_func():
         If no dataset is passed into Trainer, then return None.
     """
     session = get_session()
+    if session is None:
+        warnings.warn("`train.get_dataset_shard()` is meant to only be called "
+                      "inside a training function that is executed by "
+                      "`Trainer.run`.")
+        return
```

Comment:
consider only raising these warnings once (instead of every time a method is called)

Reply:
Done!